### ISSUE TITLE
Format of bbox

### ISSUE THREAD

**Issue by Hakuyume** [pfnet/chainercv#46]
Currently, we use a ndarray whose shape is (5,) for the format of bbox.
https://github.com/pfnet/chainercv/blob/master/chainercv/transforms/bbox/flip_bbox.py#L13

However,  it has two cons.
1. We have to use a magic index `4` when we want to access the `label_id`.
2. `label_id` is always integer. But coordinates sometimes take float values.

A structured array seems a good solution for me.
```
bbox_dtype = [('rect', 'f8', 4), ('label', 'u4')]
bbox = np.array(((1.2, 3.4, 4.5, 5.6), 3), dtype=bbox_dtype)

bbox['rect']  # array([ 1.2,  3.4,  4.5,  5.6])
bbox['label']  # array(3, dtype=uint32)
```

**Comment by yuyu2172** [comment]
> label_id is always integer. But coordinates sometimes take float values.

I understand that this is a problem.
Currently, this is handled by making a bounding box to be float type (e.g. https://github.com/pfnet/chainercv/blob/master/chainercv/datasets/pascal_voc/voc_detection_dataset.py#L100), and asking user to convert it to integer type whenever a user needs integer data.

Personally, I do not like to add additional type (`bbox_dtype = [('rect', 'f8', 4), ('label', 'u4')]`), as it would be cumbersome for users to import or copy this dtype project every time to their own projects. Also, I am not sure if this `dtype` is a standard convention in the community or not.

How about changing type of tuple returned by Dataset?
Currently, it returns

```img (CHW)  and bbox (R, 5)```

Instead,

```img (CHW),   bbox_no_label (R, 4),   and  Label (R,)```

**Comment by yuyu2172** [comment]
Also, it is important that DetectionVisReport is easy enough for users to use.
http://chainercv.readthedocs.io/en/latest/reference/extensions.html#chainercv.extensions.DetectionVisReport

**Comment by Hakuyume** [comment]
I understand why you do not want to use a structured array. I agree that an additional `dtype` is a little cumbersome.
As you suggested, to return `img (CHW), bbox_no_label (R, 4), and Label (R,)` seems a better solution.

**Comment by Hakuyume** [comment]
By defining `bbox` as a set of four coordinates, we can replace the name `bbox_no_label` with `bbox`.
This definition is natural because `bbox` of Pascal VOC does not contain class information.

**Comment by yuyu2172** [comment]
Does `img (CHW), bbox (R, 4), label (R,)` sound good to you?

**Comment by Hakuyume** [comment]
Yes. It looks good to me.

**Comment by yuyu2172** [comment]
OK.

I will change specification of bbox related functionalities.

**Comment by yuyu2172** [comment]
@furushchev
What do you think about this proposal?

**Comment by furushchev** [comment]
LGTM because:

- For resizing bounding box according to image size, `(R, 4)` would be better.
- labels are integer, values of bbox may be float.

**Comment by Hakuyume** [comment]
I find the same issue in `keypoint`.
https://github.com/pfnet/chainercv/blob/master/chainercv/transforms/keypoint/resize_keypoint.py#L4

`vaild` is always `bool` while `x` and `y` may be float.

**Comment by yuyu2172** [comment]
Solved #47.
